cpp/Platform.Data.Doublets/ILinksExtensions.h

```diff
@@ -263,16 +263,16 @@ namespace Platform::Data::Doublets
             });
             return allLinks;
         }
-    //
-    //    static Interfaces::CArray<typename TStorage::LinkAddressType>auto AllIndices<typename TStorage::LinkAddressType>(TStorage& storage, Interfaces::CArray<typename TStorage::LinkAddressType> auto&& restriction)
-    //    {
-    //        auto allIndices = List<typename TStorage::LinkAddressType>();
-    //        auto filler = ListFiller<typename TStorage::LinkAddressType, typename TStorage::LinkAddressType>(allIndices, storage.Constants.Continue);
-    //        storage.Each(filler.AddFirstAndReturnConstant, restriction);
-    //        return allIndices;
-    //    }
-    //
-    //
+
+       static Interfaces::CArray<typename TStorage::LinkAddressType>auto AllIndices<typename TStorage::LinkAddressType>(TStorage& storage, Interfaces::CArray<typename TStorage::LinkAddressType> auto&& restriction)
+        {
+            auto allIndices = List<typename TStorage::LinkAddressType>();
+            auto filler = ListFiller<typename TStorage::LinkAddressType, typename TStorage::LinkAddressType>(allIndices, storage.Constants.Continue);
+            storage.Each(filler.AddFirstAndReturnConstant, restriction);
```

Comment:
Do not use member functions and `ListFiller` too. We cannot pass member functions because they have `this` parameter in their signature, therefore the signatures do not match. Use `range for` instead of `ListFiller` and pass lambda with capture to Each. Btw the restriction must become the first argument and the handler must become the last argument in `Each` call